examples/all-clusters-app/esp32/CMakeLists.txt

```diff
@@ -20,13 +20,22 @@ cmake_minimum_required(VERSION 3.5)
 include($ENV{IDF_PATH}/tools/cmake/project.cmake)
 
 # The list of extra component dirs must be in sync with that in all-clusters-app/esp32/Makefile
+if(${IDF_TARGET} STREQUAL "esp32")
 set(EXTRA_COMPONENT_DIRS
     "${CMAKE_CURRENT_LIST_DIR}/third_party/connectedhomeip/config/esp32/components"
     "${CMAKE_CURRENT_LIST_DIR}/../../common/m5stack-tft/repo/components"
     "${CMAKE_CURRENT_LIST_DIR}/../../common/QRCode"
     "${CMAKE_CURRENT_LIST_DIR}/../../common/screen-framework"
 )
+elseif(${IDF_TARGET} STREQUAL "esp32c3")
+set(EXTRA_COMPONENT_DIRS
+    "${CMAKE_CURRENT_LIST_DIR}/third_party/connectedhomeip/config/esp32/components"
+    "${CMAKE_CURRENT_LIST_DIR}/../../common/QRCode"
+)
+endif(${IDF_TARGET} STREQUAL "esp32")
 
 project(chip-all-clusters-app)
-idf_build_set_property(CXX_COMPILE_OPTIONS "-std=c++14;-Os;-DLWIP_IPV6_SCOPES=0;-DCHIP_HAVE_CONFIG_H" APPEND)
-idf_build_set_property(C_COMPILE_OPTIONS "-Os;-DLWIP_IPV6_SCOPES=0" APPEND)
+
+idf_build_set_property(CXX_COMPILE_OPTIONS "-std=gnu++14;-Os;-DLWIP_IPV6_SCOPES=0;-DCHIP_HAVE_CONFIG_H" APPEND)
+#add -DMAXNAMLEN=7 to make compile success, but it is not safe.
```

Comment:
The option aims to fix the build issue in ESP32_TFT_library component when switching to gnu++.

Please file an issue to [ESP32_TFT_library](https://github.com/jeremyjh/ESP32_TFT_library/) project, and leave a TODO here.

Reply:
when update esp-idf version to release/v4.3, the build will fail with esp32 while std=c++14. so switch it to gnu++14